pkg/decode/decode_protobuf.go

```diff
@@ -26,6 +26,46 @@ const (
 type Protobuf struct {
 }
 
+// dnsRawNameToDotted parses a label-encoded DNS QNAME (raw bytes copied from kernel)
+// into a dotted string. Stops on NUL, compression pointer, or bounds.
+func dnsRawNameToDotted(rawI8 []int8) string {
+	// Convert to byte slice up to first NUL
+	b := make([]byte, 0, len(rawI8))
+	for i := 0; i < len(rawI8); i++ {
+		if rawI8[i] == 0 { // NUL terminator placed in kernel copy
+			break
+		}
+		b = append(b, byte(rawI8[i]))
+	}
+	if len(b) == 0 {
+		return ""
+	}
+	out := make([]byte, 0, len(b))
+	i := 0
+	first := true
+	for i < len(b) {
+		l := int(b[i])
+		if l == 0 {
+			break
+		}
+		// Stop on compression pointer (0xC0xx) since we didn't follow it in kernel
+		if (l & 0xC0) == 0xC0 {
```

Comment:
what is this 0xC0 magic number ?

Reply:
done in https://github.com/netobserv/netobserv-ebpf-agent/pull/820/commits/a4849cb1d206b1fea3ed73c72114f96f25d354b2